src/FrameBuffer.cpp

```diff
@@ -461,15 +514,24 @@ CachedTexture * FrameBuffer::getTexture(u32 _t)
 
 CachedTexture * FrameBuffer::getTextureBG(u32 _t)
 {
-	m_pTexture->scaleS = m_scale / (float)m_pTexture->realWidth;
-	m_pTexture->scaleT = m_scale / (float)m_pTexture->realHeight;
+	CachedTexture *pTexture = m_pTexture;
+
+	if (this == frameBufferList().getCurrent()) {
+		if (Context::TextureBarrier)
+			gfxContext.textureBarrier();
+		else if (Context::BlitFramebuffer)
```

Comment:
I added a check for ```Context::BlitFramebuffer``` here, so this won't work with GLES2 (GLES2 devices may still render these scenes properly, they may not, it'll just be like it is now)